sql/rpl_gtid.cc

```diff
@@ -3487,6 +3488,17 @@ my_bool Window_gtid_event_filter::has_finished()
   return m_has_stop ? m_has_passed : FALSE;
 }
 
+bool Window_gtid_event_filter::verify_final_state()
+{
+  bool is_not_final= m_has_stop && !m_has_passed;
+  if (is_not_final)
+    Binlog_gtid_state_validator::warn(stderr,
```

Comment:
`mysqlbinlog.cc`'s `static warning()` family is all duplicate code of the `Binlog_gtid_state_validator::warn()` family, except the former specifies `stderr` and has a `fflush(result_file)` step.